src/alert-prelude.c

```diff
@@ -467,11 +586,95 @@ static int PacketToDataV6(const Packet *p, const PacketAlert *pa, idmef_alert_t
  */
 static int PacketToData(const Packet *p, const PacketAlert *pa, idmef_alert_t *alert, AlertPreludeCtx *ctx)
 {
+    const char *key_js;
+    json_t *js, *s_js, *value_js;
+    char local_key[128];
+    SshState *ssh_state;
+    SSLState *ssl_state;
+
     SCEnter();
 
     if (unlikely(p == NULL))
         SCReturnInt(0);
 
+    if (p->flow != NULL) {
+        FLOWLOCK_RDLOCK(p->flow);
```

Comment:
you can drop this lock. If there is a flow (meaning p->flow != NULL) you are guaranteed it's already locked before your code was invoked. So you don't need to consider flow locking and can just use the flow.